packages/acs-ui-common/src/conditional-compilation-sample/index.tsx

```diff
@@ -204,49 +205,47 @@ export const myExtensibleSelector: MyExtensibleSelector = dummyCreateSelector(
+ * - Unconditionally add the new type
+ *   - Conditionally export the new type in the public API.
+ * - Conditionally add the new class method that uses the conditionally exposed type.
+ * - In the @public method, extract a simple helper function that detects whether the @beta method should be called.
+ *   The implementation of this function uses conditional compilation.
+ *   - Call the method if needed. Because the method and type are defined unconditionally, no conditional
+ *     compliation is needed here.
  */
 
-type DummyState = unknown;
-type DummyProps = unknown;
-function memoizedBoolean(state: DummyState, props: DummyProps) {
-  console.log(state, props);
-  return true;
+type ThisIsUnstableType = string;
+/* @conditional-compile-remove-from(stable) */
+export type { ThisIsUnstableType };
+
+class InternalImplementation {
+  public goodOldTestedFunctionality(): void {
+    /* Been doing this in stable flavor build for a decade */
+  }
+  public newMethodForUnstableFunctionality(arg: ThisIsUnstableType): string {
+    /* Unstable, but amazingly awesome magic happens here */
+    return arg;
+  }
 }
 
-function dummyCreateSelector(
-  dependencySelectors: [
-    (state: DummyState, props: DummyProps) => boolean,
-    (state: DummyState, props: DummyProps) => boolean,
-    (state: DummyState, props: DummyProps) => boolean
-  ],
-  func: (a: boolean, b: boolean, c: boolean) => { memoizedA: boolean; memoizedB: boolean; memoizedC: boolean }
-): (state: DummyState, props: DummyProps) => { memoizedA: boolean; memoizedB: boolean; memoizedC: boolean };
-function dummyCreateSelector(
-  dependencySelectors: [
-    (state: DummyState, props: DummyProps) => boolean,
-    (state: DummyState, props: DummyProps) => boolean,
-    (state: DummyState, props: DummyProps) => boolean
-  ],
-  func: (a: boolean, b: boolean, c: boolean) => { memoizedA: boolean; memoizedB: boolean }
-): (state: DummyState, props: DummyProps) => { memoizedA: boolean; memoizedB: boolean };
-function dummyCreateSelector(
-  dependencySelectors: [
-    (state: DummyState, props: DummyProps) => boolean,
-    (state: DummyState, props: DummyProps) => boolean,
-    (state: DummyState, props: DummyProps) => boolean
-  ],
-  func: (a: boolean, b: boolean, c: boolean) => unknown
-) {
-  return (state: DummyState, props: DummyProps) => {
-    return func(
-      dependencySelectors[0](state, props),
-      dependencySelectors[1](state, props),
-      dependencySelectors[2](state, props)
-    );
-  };
+export function StableAPIThatGainsNewUnstableBehavior(): void {
+  const impl = new InternalImplementation();
+  impl.goodOldTestedFunctionality();
+  if (shouldIncludeUnstableFeature()) {
+    const arg: ThisIsUnstableType = 'getThisFromUser';
+    impl.newMethodForUnstableFunctionality(arg);
+  }
+}
+
+function shouldIncludeUnstableFeature(): boolean {
+  /* @conditional-compile-remove-from(stable) */
+  return true;
```

Comment:
Can we double check if ts compiler or linter will throw error for this double return setences?

Reply:
I checked that neither does. CI is also green (runs both build and lint).